### PR TITLE
88 upgrade nameservice

### PR DESCRIPTION
**Pull request by sirpy**
# Description

About #:88

### CODE REVIEW

**Review by aliberkhsl — COMMENTED** [review]
I did some changes can you review them?

**Review by sirpy — COMMENTED** [review]
@aliberkhsl 
If the tests didnt find these calculation mistakes it means they are not good! or we are not testing at all!

**Review by sirpy — COMMENTED** [review]
@aliberkhsl 
comment with explanation to test calculations...

File: test/staking/StakingRewards.test.ts [hunk 1]
```diff
@@ -1402,11 +1402,11 @@ describe("StakingRewards - staking with cDAI mocks and get Rewards in GoodDollar
   it("it should calculate price of spent gas in DAI properly", async () => {
     const gasAmount = BN.from("1100000"); // 1.1M
     const gasPrice = await gasFeeOracle.latestAnswer(); // returns 25 gwei
-    const daiToEthRate = await daiEthOracle.latestAnswer(); // returns  0.000341481428801721 and we accept 1$ = 1DAI
+    const daiToEthRate = await daiEthOracle.latestAnswer(); // returns  0.000341481428801721
     const calculatedResult = gasPrice
       .mul(BN.from("10").pow(18))
-      .div(daiToEthRate)
-      .mul(gasAmount); // Result should be around 80.53$
+      .div(daiToEthRate) // gasPrice in ETH so 18 decimals and DAI/ETH rate is in also 18 decimals so in order to 18 decimals we multiply with 1e18
```
Comment (sirpy) [inline test/staking/StakingRewards.test.ts:1408]: there's no multiplication by 1e18 ehre

File: test/staking/StakingRewards.test.ts [hunk 2]
```diff
@@ -1418,11 +1418,11 @@ describe("StakingRewards - staking with cDAI mocks and get Rewards in GoodDollar
     const gasAmount = BN.from("1100000"); // 1.1M
     const gasPrice = await gasFeeOracle.latestAnswer(); // returns 25 gwei
     const daiToEthRate = await daiEthOracle.latestAnswer(); // returns  0.000341481428801721 and we accept 1$ = 1DAI
-    const gasPriceInDAI = gasPrice.mul(BN.from("10").pow(18)).div(daiToEthRate);
+    const gasPriceInDAI = gasPrice.mul(BN.from("10").pow(18)).div(daiToEthRate); // gasPrice in ETH so 18 decimals and DAI/ETH rate is in also 18 decimals so in order to 18 decimals we multiply with 1e18
     const gasPriceInCdai = gasPriceInDAI
       .div(BN.from("10").pow(10))
       .mul(BN.from("10").pow(28))
```
Comment (sirpy) [inline test/staking/StakingRewards.test.ts]: not explained

File: test/staking/StakingRewards.test.ts (hunk 1 above)
Comment (aliberkhsl) [reply test/staking/StakingRewards.test.ts:1408]: .mul(BN.from("10").pow(18))

Comment (aliberkhsl) [reply test/staking/StakingRewards.test.ts:1408]: Ok I'll change comments step by step

Comment (aliberkhsl) [reply test/staking/StakingRewards.test.ts]: changed it

File: test/staking/StakingRewards.test.ts (hunk 2 above)
Comment (aliberkhsl) [reply test/staking/StakingRewards.test.ts]: added explanations

**Review by aliberkhsl — APPROVED** [review]

**Pull request merged**